packages/react-native-builder-bob/src/utils/patchCodegen.ts

```diff
@@ -0,0 +1,89 @@
+import fs from 'fs-extra';
+import path from 'path';
+
+/**
+ * Currently, running react-native codegen generates java files with package name `com.facebook.fbreact.specs`.
+ * This is a known issue in react-native itself.
+ * You can find the relevant line here: https://github.com/facebook/react-native/blob/dc460147bb00d6f912cc0a829f8040d85faeeb13/packages/react-native/scripts/codegen/generate-artifacts-executor.js#L459.
+ * To workaround, this function renames the package name to the one provided in the codegenConfig.
+ * @throws if codegenConfig.outputDir.android or codegenConfig.android.javaPackageName is not defined in package.json
+ * @throws if the codegenAndroidPath does not exist
+ */
+export async function patchCodegen(projectPath: string) {
+  const packageJsonPath = path.resolve(projectPath, 'package.json');
+  const packageJson = await fs.readJson(packageJsonPath);
+
+  let codegenAndroidPath: string | undefined =
+    packageJson.codegenConfig.outputDir.android;
+  if (!codegenAndroidPath) {
+    throw new Error(
+      'You need to define codegenConfig.outputDir.android in your package.json'
```

Comment:
Let's add some links to the documentation of how `codegenConfig` should look like

Reply:
idea: https://github.com/react-navigation/react-navigation/blob/fdbfe7608c3c9ba4c2c3c95dcc5a6501ec97a056/packages/core/src/useFocusEffect.tsx#L19